climada/hazard/base.py

```diff
@@ -1457,6 +1457,18 @@ def select(self, event_names=None, event_id=None, date=None, orig=None,
                                    dt.datetime.fromordinal(haz.date.min()).year) + 1
             haz.frequency = haz.frequency * year_span_old / year_span_new
 
+        # Check if new fraction is zero everywhere
+        if self.fraction.nnz != 0 and haz.fraction.nnz == 0:
```

Comment:
Just a small efficiency comment: `crs_matrix.nnz` is equivalent to `np.count_nonzero(a.toarray())`. Thus, it first projects to an array and then counts. This is rather costly for large matrices, and here is also not needed. It is much faster to check `self.fraction.size != 0` and `haz.fraction.size == 0`. The only downside is IF the fraction matrix is not well handled (i.e., some explicit 0s are present).

Reply:
> crs_matrix.nnz is equivalent to np.count_nonzero(a.toarray())

This is incorrect. `nnz` returns the number of explicitly stored values, including explicitly stored zeros. `np.count_nonzero(a.toarray())` would count all zeros in the assembled dense matrix.

> Thus, it first projects to an array and then counts.

As far as I can see in the scipy source code, this is also incorrect. Where do you take this information from?

> The only downside is IF the fraction matrix is not well handled (i.e., some explicit 0s are present).

This is indeed correct. We can call `csr_matrix.eliminate_zeros` during `select` to make sure this does not happen.

Reply:
ooo you are right, sorry, I misread one element of the documentation, sorry! I was so confused now trying to find where `.nnz` is defined exactly. In the base class of sparse, `.size` and `.nnz` are identical. Anyway, it seems I was mistaken. I thought `.nzz` is an inefficient way of checking whether a matrix is empty or not. But that was wrong. Sorry for the confusion!


> This is indeed correct. We can call csr_matrix.eliminate_zeros during select to make sure this does not happen.

Not sure if we need to do that since a proper climada hazard should not contain explicit 0s... but probably it does not hurt. Then we have to do it for intensity and for fraction right?

Reply:
We technically only need it for `fraction` here, because the values of `intensity` do not matter during selection 🤷

Reply:
After personal discussion: Go with `_get_fraction` although the linter will complain 😵

Reply:
When implementing this for `fraction`, I noticed that the csr matrix automatically eliminates zeros when you do "regular" value assignment. Only when manipulating the attributes directly, the matrix might end up in an inconsistent state. So, I think there is no action needed for `intensity`, as few users will do such a thing.

Example:
```python
fraction[1, 2] = 0  # This zero will *not* be stored explicitly

fraction.data[0] = 0  # This zero *will* be stored explicitly, therefore...
fraction.eliminate_zeros()  # ...is needed.
```

Reply:
Good to know!